### PR TITLE
PDE-2333: Fix duplicate API Key headers part 2

### PR DESCRIPTION
**Pull request by eliangcs**
Turns out #364 only fixes part of the issue. This PR fixes another edge case that can lead to duplicate API Key headers.

### CODE REVIEW

File: packages/legacy-scripting-runner/test/example-app/index.js [hunk 1]
```diff
@@ -317,6 +317,15 @@ const legacyScriptingSource = `
         return bundle.request;
       },
 
+      movie_pre_poll_double_headers: function(bundle) {
```
Comment (eliangcs) [inline packages/legacy-scripting-runner/test/example-app/index.js:320]: Before calling `movie_pre_poll_double_headers`, `bundle.request.headers` has a `X-Api-Key: one` header.

File: packages/legacy-scripting-runner/test/example-app/index.js [hunk 2]
```diff
@@ -317,6 +317,15 @@ const legacyScriptingSource = `
...
+        bundle.request.url = '${HTTPBIN_URL}/get';
+        bundle.request.headers = {
+          'x-api-key': 'two',
+          'X-API-KEY': 'three'
+        };
+        return bundle.request;
```
Comment (eliangcs) [inline packages/legacy-scripting-runner/test/example-app/index.js:326]: By the time `movie_pre_poll_double_headers` returns, `bundle.request.headers` only has two entries: `x-api-key: two` and `X-API-KEY: three`.

We expect the final headers should only contain one entry `x-api-key: three`.

File: packages/legacy-scripting-runner/index.js [hunk 3]
```diff
@@ -295,22 +295,22 @@ const cleanHeaders = (headers) => {
...
   const lowerToFirstKeys = {};
 
-  const newHeaders = {};
-  Object.entries(headers).forEach(([k, v], index) => {
+  const newHeaders = Object.entries(headers).reduce((result, [k, v]) => {
     const lowerKey = k.toLowerCase();
     if (lowerToFirstKeys[lowerKey] === undefined) {
       lowerToFirstKeys[lowerKey] = k;
     }
     const firstKey = lowerToFirstKeys[lowerKey] || k;
```
Comment (szchenghuang) [inline packages/legacy-scripting-runner/index.js:310]: @eliangcs It seems `const firstKey = lowerToFirstKeys[lowerKey]` is enought?

**Review by szchenghuang — APPROVED** [review]

File: packages/legacy-scripting-runner/index.js (hunk 3 above)
Comment (eliangcs) [reply packages/legacy-scripting-runner/index.js]: @szchenghuang you're right. Removed that in ad2b37a. Thanks!
